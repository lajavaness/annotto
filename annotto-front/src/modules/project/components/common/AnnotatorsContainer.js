import { Tooltip } from 'antd'
import { isEmpty } from 'lodash'
import { useTranslation } from 'react-i18next'
import PropTypes from 'prop-types'
import React, { useMemo } from 'react'

import * as Styled from './__styles__/AnnotatorsContainer.styles'

const AnnotatorsContainer = ({ annotators, currentUser }) => {
	const { t } = useTranslation('project')

	const resolvedAnnotators = useMemo(
		() => (!isEmpty(annotators) ? [currentUser, ...annotators.filter(({ email }) => email !== currentUser.email)] : []),
		[currentUser, annotators]
	)

	return (
		<Styled.Root>
			{!isEmpty(resolvedAnnotators) ? (
				resolvedAnnotators.map(
					({ email, firstName, lastName }) =>
						firstName &&
						lastName && (
							<Tooltip placement="top" title={`${firstName} ${lastName}`} key={email}>
								<Styled.Avatar shape="circle" size={35} $isSelected={email === currentUser.email}>
									{`${firstName.charAt(0)}${lastName.charAt(0)}`}
								</Styled.Avatar>
							</Tooltip>
						)
				)
			) : (
				<Styled.Span>{t('project:annotators.noAnnotators')}</Styled.Span>
			)}
		</Styled.Root>
	)
}

export default AnnotatorsContainer

const User = PropTypes.shape({
	/** Defines a machine-readable key that identifies the user. */
	_id: PropTypes.string,
	/** Defines the firstName of an user. */
	firstName: PropTypes.string,
	/** Defines the lastName of an user. */
	lastName: PropTypes.string,
	/** Defines the email of an user. */
	email: PropTypes.string,
})

AnnotatorsContainer.propTypes = {
	/** Defines annotators to display. */
	annotators: PropTypes.arrayOf(User),
	currentUser: User,
}

AnnotatorsContainer.defaultProps = {
	annotators: [],
	currentUser: null,
}
