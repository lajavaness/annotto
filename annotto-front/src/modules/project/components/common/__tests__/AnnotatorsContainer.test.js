import { ThemeProvider } from 'styled-components'
import { render } from '@testing-library/react'
import React from 'react'

import theme from '__theme__'

import AnnotatorsContainer from 'modules/project/components/common/AnnotatorsContainer'

const defaultProps = {}
const getInstance = (props = {}) => (
	<ThemeProvider theme={theme}>
		<AnnotatorsContainer {...defaultProps} {...props} />
	</ThemeProvider>
)

describe('AnnotatorsContainer', () => {
	it('matches snapshot', () => {
		const { asFragment } = render(getInstance())
		expect(asFragment()).toMatchSnapshot()
	})
})
