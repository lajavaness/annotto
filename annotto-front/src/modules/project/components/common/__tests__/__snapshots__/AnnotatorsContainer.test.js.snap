// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnnotatorsContainer matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-gsBqHK gsESNF"
  >
    <div
      class="sc-kmATbt cDljgH"
    >
      project:annotators.noAnnotators
    </div>
  </div>
</DocumentFragment>
`;
